File: java/src/main/java/com/android/frame/mvp/AATest/adapter/NewsListAdapter.java
package com.android.frame.mvp.AATest.adapter;

import android.content.Context;
import android.widget.ImageView;

import com.android.base.BaseApplication;
import com.android.frame.mvp.AATest.bean.NewsListBeanMvp;
import com.android.java.R;
import com.android.widget.RecyclerView.LoadMoreAdapter;
import com.android.widget.RecyclerView.ViewHolder;
import com.bumptech.glide.Glide;

import java.util.List;

/**
 * Created by xuzhb on 2020/1/8
 * Desc:
 */
public class NewsListAdapter extends LoadMoreAdapter<NewsListBeanMvp> {

    public NewsListAdapter(Context context, List<NewsListBeanMvp> dataList) {
        super(context, dataList, R.layout.item_test_load_more);
    }

    @Override
    protected void bindData(ViewHolder holder, NewsListBeanMvp data, int position) {
        ImageView imageIv = holder.getView(R.id.image_iv);
        Glide.with(BaseApplication.getInstance()).load(data.getImage()).into(imageIv);
        holder.setText(R.id.title_tv, data.getTitle()).setText(R.id.time_tv, data.getPasstime());
    }
}

File: kotlin/src/main/java/com/android/frame/mvp/AATest/adapter/NewsListAdapter.kt
package com.android.frame.mvp.AATest.adapter

import android.content.Context
import android.widget.ImageView
import com.android.base.BaseApplication
import com.android.basicproject.R
import com.android.frame.mvp.AATest.bean.NewsListBeanMvp
import com.android.widget.RecyclerView.BaseAdapter
import com.android.widget.RecyclerView.ViewHolder
import com.bumptech.glide.Glide

/**
 * Created by xuzhb on 2020/1/4
 * Desc:
 */
class NewsListAdapter(context: Context, list: MutableList<NewsListBeanMvp>) :
    BaseAdapter<NewsListBeanMvp>(context, list, R.layout.item_test_load_more) {
    override fun bindData(holder: ViewHolder, data: NewsListBeanMvp, position: Int) {
        val imageIv: ImageView = holder.getView(R.id.image_iv)!!
        Glide.with(BaseApplication.instance).load(data.image).into(imageIv)
        holder.setText(R.id.title_tv, data.title).setText(R.id.time_tv, data.passtime)
    }
}